### PR TITLE
Expose lnurlAuth & generate Dart bindings

### PR DESCRIPTION
**Pull request by erdemyerebasmaz**
- Make LnUrlAuthCallbackStatus compatible with Dart binding
- Expose perform_lnurl_auth via lnurl_auth in bindings
- Add documentation for lnurl auth
  - Change or add additional information as you like
- Detailed lnurlWithdraw's documentation

### CODE REVIEW

File: libs/sdk-core/src/lnurl/auth.rs [hunk 1]
```diff
@@ -9,9 +10,35 @@ use reqwest::Url;
 use std::str::FromStr;
 use std::sync::Arc;
 
-pub type LnUrlAuthCallbackStatus = LnUrlWithdrawCallbackStatus;
+pub(crate) mod model {
```
Comment (roeierez) [inline libs/sdk-core/src/lnurl/auth.rs:13]: Does it mean that now we have duplication of this type? Also in the models.rs?

Comment (roeierez) [reply libs/sdk-core/src/lnurl/auth.rs:13]: If there isn't any difference between them can't we use one type for both? LnUrlCallbackStatus maybe?
@ok300 what do you think?

Comment (erdemyerebasmaz) [reply libs/sdk-core/src/lnurl/auth.rs]: > If there isn't any difference between them can't we use one type for both? LnUrlCallbackStatus maybe?
> @ok300 what do you think?

Done https://github.com/breez/breez-sdk/pull/150/commits/a6a804e63aecc203e2ad4159c673672efdbcf123

**Comment by roeierez** [comment]
@erdemyerebasmaz I think we need also to implement the react native changes

File: libs/sdk-core/src/lnurl/auth.rs (hunk 1 above)
Comment (roeierez) [reply libs/sdk-core/src/lnurl/auth.rs]: @dangeross this change introduce a change in the lnurl operations return type. It unified the result into one structure: `LnUrlCallbackStatus` that has the same structure for all lnurl operations.
Would you like to review that before we are moving forward?

**Comment by erdemyerebasmaz** [comment]
@dangeross Latest changes can't be seen on this PR right now due to a git issue. Meanwhile you can review react native changes here https://github.com/breez/breez-sdk/commit/2c8db6f9fd6e3e5e75cadf921f2d6a867c83a951

File: libs/sdk-core/src/lnurl/auth.rs (hunk 1 above)
Comment (dangeross) [reply libs/sdk-core/src/lnurl/auth.rs]: @roeierez Sure thing, I'll take a look now

**Review by dangeross — APPROVED** [review]

**Review by roeierez — APPROVED** [review]

**Pull request merged**